api/v1beta1/cluster_types.go

```diff
@@ -184,6 +194,25 @@ type MachineDeploymentTopology struct {
 	Variables *MachineDeploymentVariables `json:"variables,omitempty"`
 }
 
+// MachineHealthCheckTopology defines a MachineHealthCheck for a group of machines.
+type MachineHealthCheckTopology struct {
+	// Enable controls if a MachineHealthCheck should be created for the target machines.
+	// If false:
+	//		No MachineHealthCheck will be created.
+	// If not set:
+	//		A MachineHealthCheck will be created if it is defined here or in the associated ClusterClass.
+	//		If no MachineHealthCheck is defined then none will be created.
+	// If true:
+	//		A MachineHealthCheck is guaranteed to be created. Cluster validation will block if `enable` is
+	//		true and no MachineHealthCheck definition is available.
+	// +optional
+	Enable *bool `json:"enable,omitempty"`
+
+	// MachineHealthCheckClass defines a MachineHealthCheck for a group of machines.
+	// If specified, it overrides the MachineHealthCheckClass defined in ClusterClass.
```

Comment:
```suggestion
	// If specified (any field is set), it entirely overrides the MachineHealthCheckClass defined in ClusterClass.
```
let's make it super clean there is no merge behaviour